### PR TITLE
Aviah/docs

### PR DESCRIPTION
**Pull request by Aviah**
### What was wrong?
Detailed docs 

### How was it fixed?
docs

#### Cute Animal Picture

duck_ _ducklings_morning_walk

### CODE REVIEW

File: docs/dev_cycle.intro.rst [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+The purpose of this tutorial is to go one step beyond the common "Hello World" Greeter example,
+to the entire developement cycle of an Ethereum smart contract. We will try to unpack
+the confusing issues, clear up the mystery, and have some fun when you dive into blockchain and contracts developemnt.
+
+The tools we will use are Populus and Web3.py
+
+.. note::
+
+    Web3 in general is the client side API that let you intreacts with the blockchain. Populus is a
```
Comment (pipermerriam) [inline docs/dev_cycle.intro.rst]: s/intreacts/interact/

File: docs/dev_cycle.intro.rst [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+.. note::
+
+    Web3 in general is the client side API that let you intreacts with the blockchain. Populus is a 
+    development framework, that is built on top of Web3. If you are into javascript, you can use the
+    `Truffle javascript framework <http://truffleframework.com/>`_, and Web3.js which ships with the
+    ``geth`` client. If you prefer Python, then Populus and Web3.py are your friends.
+    
+
+We assume that your read 1-2 intros about Ethereum and the blockchain, and know Python.
```
Comment (pipermerriam) [inline docs/dev_cycle.intro.rst]: Maybe: *"This tutorial assumes you have some very basic understanding of the ethereum blockchain as well as with the Python programming language"*

File: docs/dev_cycle.intro.rst [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+You don't need the complex math of the elliptic curves, but to get a grasp of the basic concepts, and the basic idea: A system that prevents bad behaviour not by moral rules, but 
+by incentives. Incentinve that make honest behaviour *more* profitable (let this bold concept sink in for a moment).
+
+Two good intros:
+
+* `A 101 Noob Intro to Programming Smart Contracts on Ethereum <https://medium.com/@ConsenSys/a-101-noob-intro-to-programming-smart-contracts-on-ethereum-695d15c1dab4>`_, from Consensys
+
+* `A Gentle Intorduction to Ethereum <https://bitsonblocks.net/2016/10/02/a-gentle-introduction-to-ethereum/>`_, from bitsonblocks
+
+And to go a little deeper:
```
Comment (pipermerriam) [inline docs/dev_cycle.intro.rst]: Suggest removing the following links, or relocating them to the end of this tutorial as to avoid potential confusion and extra superfluous information.

File: docs/dev_cycle.intro.rst [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+* Writing the contract
+* Testing, fixing bugs
+* Deployment to a local chain, make sure everything works
+* Deployment to testnet,
+* Finally deployment to mainnet which will cost real gas
+* Interaction with the contract on the blockchain.
+
+
+Glossery
```
Comment (pipermerriam) [inline docs/dev_cycle.intro.rst]: s/Glossery/Glossary/

File: docs/dev_cycle.intro.rst [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+--------
+
+Just a succint reference, as a reminder if you need it during the tutorial (in a "chronological" order)
```
Comment (pipermerriam) [inline docs/dev_cycle.intro.rst]: s/succint/succinct

File: docs/dev_cycle.intro.rst [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+and data. The data is used to run a contract if the account has one.
+
+**Why the private key and the public key?** The keys mechanism can confirm that the transaction was indeed authorised by the account owner, 
+that claims to sent it. 
+
+**Block**: A group of transactions
+
+**Mining**: Bundling a group of transactions into a block
+
+**Why mining is hard?** Because the miner needs to bundle transactions with additional input that requires significant
```
Comment (pipermerriam) [inline docs/dev_cycle.intro.rst]: Suggest removing this as it's only applicable to POW and also kind of superfluous.

File: docs/dev_cycle.intro.rst [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+**geth**: The official implemntation of an Ethereum blockchain node, written in Go
+
+**gas**: The price that users pay to run computational actions on the blockchain (deploying a new contract, send money, run a contract function, storage)
+
+**mainnet**: The Ethereum blockchain
+
+**testnet**: An Ethereum blockchain for testing. It behaves exactly as mainnet, but you don't use real to pay for the Ether and the gas
+
+**Local chain**: A blockchain that runs localy, has it's own blocks, and does not sync to any other blockchain. Useful for development
```
Comment (pipermerriam) [inline docs/dev_cycle.intro.rst]: All of these might be well suited for a glossary section of the docs that we can the link to when we use the words in tutorials?

File: docs/dev_cycle.part-01.rst [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+in terms of Ether, but similar donations in terms of USD.
+
+In other words, two donations of say $30 will have different amounts in ETH if the exchange rate changed between
+the donations. As a simple solution, we will ask donators to provide the effective ETH/USD exchange rate when they send their (hopefully generous) donations.
+
+Here is the new contract code:
+
+.. code-block:: solidity
+
+    pragma solidity ^0.4.0;
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: 0.4.0 feels old.  How about using the minimum version of solidity that populus is about to support?

File: docs/dev_cycle.part-01.rst [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+
+    /// TUTORIAL CONTRACT DO NOT USE IN PRODUCTIO
+    /// @title Donations collecting contract
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: Sphinx has a directive called a *literal include* which lets you include the contents of a file inline.  This would potentially be a good tool for these inlined source files.  We could include them in the `ASSETS_DIR`, have some tests that verified that they compiled, and then just inline them here.  That would help make sure these tutorials stayed functional.

File: docs/dev_cycle.part-01.rst [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+
+    contract Donator {
+
+      uint public donations_total;
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: Can you change the naming conventions used here to follow the solidity style guide.

http://solidity.readthedocs.io/en/develop/style-guide.html

File: docs/dev_cycle.part-01.rst [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+
+**Data types**:
+Since we are dealing with numbers, the only data type we use here is ``uint``, unsigned integer. The ``int`` and ``uint``
+are declated in steps of 8 bits, ``unint8``, ``uint16`` etc. When the bits indicator is omitted, like ``int`` or ``uint``, the compiler will
+assumes ``uint256``.
+
+.. note:: 
+
+    If you know in advance the the maximum size of a variable, better to limit the type and save the gas of extra 
+    memory or storage.
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: This is not entirely accurate.  Solidity only saves memory and storage in certain situations, and often, using smaller types means higher gas costs because solidity has to do the conversions.

File: docs/dev_cycle.part-01.rst [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+    "chains": {
+      "horton": {
+        "chain": {
+          "class": "populus.chain.ExternalChain"
+        },
+        "web3": {
+          "provider": {
+            "class": "web3.providers.ipc.IPCProvider",
+          "settings": {
+            "ipc_path":"/home/mary/projects/donations/horton/chain_data/geth.ipc"
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: Can this be a relative path?  Seems more portable.

File: docs/dev_cycle.part-01.rst [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+
+    If you know in advance the the maximum size of a variable, better to limit the type and save the gas of extra 
+    memory or storage.
+    
+As of version 0.4.17 Solidity does *not* support decimal point types. If you need decimal point, you will have to manauly handle
+the fixed point calculations with integers. For the sake of simplicty, the example uses only ints.
+
+    
+**Constructor**:
+The function ``function Donator()`` is a constructor. A constructor function's name is always identical to the contract's name.
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: Suggest change to:

> A constructor function's name **must** be identical to the contract's name.

File: docs/dev_cycle.part-01.rst [hunk 14]
```diff
@@ -0,0 +1,288 @@
...
+**Modifiers**:
+``modifier money_sent() { if (!msg.value > 0) throw; _; }``. The term "modifier" is a bit confusing. 
+A modifier of a function is  *another* function that injects, or modifies, code, typically to verify some pre-existing condition.
+Since the donate function uses the modifier ``function donate(uint usd_rate) public payable money_sent {...}``,
+then ``money_sent`` will run *before* ``donate``. The code in ``donate`` will run only if ``msg.value > 0``, and make sure
+that the ``donations_count`` does not increase by a zero donation.
+
+.. note::
+    
+    The modifier syntax uses ``_;`` to tell solidity where to insert the *modified* function.
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: Suggest changing to:

> where to insert the function body.

File: docs/dev_cycle.part-01.rst [hunk 15]
```diff
@@ -0,0 +1,288 @@
...
+    We can of course check the include the modifier condition the original function, but a declared modifier is handy
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: Suggest change to:

> The conditional checks from the ``money_sent`` modifier could be embedded into the function body itself.  Modifiers are useful ways to make these conditional checks reusable.

File: docs/dev_cycle.part-01.rst [hunk 16]
```diff
@@ -0,0 +1,288 @@
...
+
+      uint public donations_total;
+      uint public donations_usd;
+      uint public donations_count;
+      uint public default_usd_rate;
+
+      function Donator() {
+        default_usd_rate = 350;
+      }
+      modifier money_sent() { if (!msg.value > 0) throw; _; }
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: Suggest renaming to `non_zero_value`?

File: docs/dev_cycle.part-01.rst [hunk 17]
```diff
@@ -0,0 +1,288 @@
...
+that the ``donations_count`` does not increase by a zero donation.
+
+.. note::
+    
+    The modifier syntax uses ``_;`` to tell solidity where to insert the *modified* function.
+    We can of course check the include the modifier condition the original function, but a declared modifier is handy
+    when you want to use the same pre-condition validation in more than one function.
+
+**Fallback**:
+The weired function without a name, ``function () payable {...}``, is the "fallback". It calls ``donate``, so when somebody just
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: suggest removing the word *weirded*(both not a word and I think it takes away from legability)

File: docs/dev_cycle.part-01.rst [hunk 18]
```diff
@@ -0,0 +1,288 @@
...
+send Ether to the contract address without explicitly call ``donate``, we can still accept the Ether. A fallback function
+is what the contract runs when called *without an explicit function name*. This happens (a) when you call a contract
+with  ``address.call``, and (b) when just send just Ether, in a transaction that don't call anything.
+
+.. note::
+
+    If a contract has a fallback function, and you should assume it has one, then just sending
```
Comment (pipermerriam) [inline docs/dev_cycle.part-01.rst]: This would be more accurate to say that:

> Any transaction or sending of ether to an address with code will result in it's code being invoked.

File: docs/dev_cycle.part-02.rst [hunk 19]
```diff
@@ -0,0 +1,174 @@
...
+
+.. contents:: :local:
+
+Compile
+-------
+
+Solidity compiles the source to EVM bytecode, the operations codes that will actually
+run on the blockchain.
+
+To compile, first check that solidity works on your machine. Try to get the latest version.
```
Comment (pipermerriam) [inline docs/dev_cycle.part-02.rst]: This might be a good place to drop in a snippet about using `py-solc` to install a specific version of `solc`

File: docs/dev_cycle.part-02.rst [hunk 20]
```diff
@@ -0,0 +1,174 @@
...
+  ^-^
+  
+      > command: `solc --standard-json`
+      > return code: `0`
+      > stderr:
+      {"contracts":{},"errors":[{"component":"general","formattedMessage":"contracts/Donator.sol:21:5: DeclarationError: Identifier not found or not unique.\n    uin in_usd = msg.value * usd_rate;\n    ^-^\n","message":"Identifier not found or not unique.","severity":"error","type":"DeclarationError"}],"sources":{}}
+  
+      > stdout:
+
+What's that? actually it's not that bad. You can ignore the Python traceback, which is just the Populus call stack until the actual
```
Comment (pipermerriam) [inline docs/dev_cycle.part-02.rst]: > You can ignore the Python traceback

Maybe this would be a good time to have `populus` wrap that and not dump the whole traceback when it's a `SolcError`?

File: docs/dev_cycle.part-02.rst [hunk 21]
```diff
@@ -0,0 +1,174 @@
...
+
+The fixed line should be:
+
+.. code-block:: solidity
+
+  uint in_usd = msg.value * usd_rate;
+  
+.. note::
+
+  Try the `online IDE <https://remix.ethereum.org>`_ , which has great interactive compiler and web-form like interface
```
Comment (pipermerriam) [inline docs/dev_cycle.part-02.rst]: Change "online IDE" to "Remix IDE" so that it's referenced by name (people know it better as Remix)

File: docs/dev_cycle.part-02.rst [hunk 22]
```diff
@@ -0,0 +1,174 @@
...
+  > Wrote compiled assets to: build/contracts.json
+  
+Nice. The two contracts are now compiled. Take a look at the file that Populus just added,
+``build/contracts.json``. The file saves some of the compilers output, which will be useful later.
+
+.. note::
+  
+    Compilation creates ``bytecode`` and ``bytecode_runtime``. The ``bytecode`` contains the ``bytecode_runtime``,
+    as well as additional code. The additional code is required to deploy the runtime, but once deployed
+    the runtime *is* the contract on the blockchain.
```
Comment (pipermerriam) [inline docs/dev_cycle.part-02.rst]: Change to:

> the runtime *is* the bytecode that will be present on the blockchain.

File: docs/dev_cycle.part-03.rst [hunk 23]
```diff
@@ -0,0 +1,194 @@
...
+  
+    py.test collects all the tests that follow
+    it's `naming conventions <https://pytest.readthedocs.io/en/reorganize-docs/new-docs/user/naming_conventions.html>`_
+    
+We don't need the Greeter example for this project, so delete it:
+
+.. code-block:: bash
+
+  $ rm contracts/Greeter 
+  4 rm tests/test_greeter.py
```
Comment (pipermerriam) [inline docs/dev_cycle.part-03.rst]: s/4/$/

File: docs/dev_cycle.part-03.rst [hunk 24]
```diff
@@ -0,0 +1,194 @@
...
+
+However, the term *contract* can be confusing, since it's used for several different things:
+
+#. A bytecode program that is saved on the blockchain
+#. A ``contract`` code in a Solidity source file
+#. A Web3 ``contract`` object
+
+Luckily, it can be even more confusing. You may have a Solidity source file of the ``Donator`` contract. If you deployed it 
+to ``testnet``, then we have another "contract", the code that sits on ``testnet``. When you deploy to ``mainnet``,
+which is another blockchain, we now have three contracts: the source file, the bytecode on ``testnet``, and the bytecode on ``mainnet``.
```
Comment (pipermerriam) [inline docs/dev_cycle.part-03.rst]: > we now have three contracts: the source file, the bytecode on ``testnet``, and the bytecode on ``mainnet``.

Might be easier to read if this was a numbered or bullet-ed list.

File: docs/dev_cycle.part-03.rst [hunk 25]
```diff
@@ -0,0 +1,194 @@
...
+not need the solidity source, since the bytcode is alreadt compiled and deployed. It does need the ABI,  the functions 
+and arguments structure of the *bytecode* contract's interface, and the address of this *bytecode* contract
+on the mainnet.
+
+.. note::
+  To deploy a *new* ``Donator`` contract, maybe to another blockchain, or to another address on the same blockchain,
+  we also need a Web3 contract *object*, but this time the object does require a compiled
+  "contract"
+
+Actually when we say ""contract", we deal with three different things:
```
Comment (pipermerriam) [inline docs/dev_cycle.part-03.rst]: This sentence is awkward.  I feel like this list and information is already present a few lines up.

File: docs/dev_cycle.part-03.rst [hunk 26]
```diff
@@ -0,0 +1,194 @@
...
+
+Get the blockchain
+''''''''''''''''''
+
+Another bonus is the ``chain``, at ``def test_default_usd_rate(chain)``. It gives the test function a Python object 
+that corresponds to a running blockchain, the ``tester`` blockchain.
+Reminder: The ``tester`` chain is ephemeral, saved only in memory, and will reset on every test run.
+
+The ``chain`` argument is a py.test *fixture*: in py.test world it's a special argument that the test function can accept.
+You don't have to declare or assign it, it's just ready and available for your function.
```
Comment (pipermerriam) [inline docs/dev_cycle.part-03.rst]: Maybe change to:

> `py.test` automatically *injects* these arguments into your test functions for you.

And maybe add a link to the pytest documentation on fixtures for *further reading*

File: docs/dev_cycle.part-03.rst [hunk 27]
```diff
@@ -0,0 +1,194 @@
...
+
+Run the First Test: Public State Variable
+-----------------------------------------
+
+We are ready for the first test: we have a test function that runs the ``tester`` chain, and using ``get_or_deploy_contract('Donator')``
+it compiles ``Donator.sol``, deploys it to the ``tester`` chain, gets a Python contract object that wraps the actual contract's
+bytecode on the chain, and assign this object to a variable, ``donator``.
+
+Once we have the ``donator`` contract as a Python object, we can call any function of this contract. You get the *contract's*
+interface with ``call()``. Reminder: ``call`` behaves exactly as a transaction, but does not alter state. It's like a "dry-run".
```
Comment (pipermerriam) [inline docs/dev_cycle.part-03.rst]: Maybe change *It's like a dry run* to:

> `call()` is good for reading data from contracts, but it **cannot** be used to change any contract state.

File: docs/dev_cycle.part-03.rst [hunk 28]
```diff
@@ -0,0 +1,194 @@
...
+
+The first test important line is:
```
Comment (pipermerriam) [inline docs/dev_cycle.part-03.rst]: I don't understand what this is trying to say.

File: docs/dev_cycle.part-03.rst [hunk 29]
```diff
@@ -0,0 +1,194 @@
...
+  function Donator() {
+    default_usd_rate = 350;
+  }
+  ...
+  
+To recap, ``default_usd_rate`` is a ``public`` variable, hence the compiler automatically created
+an accessor function, a "get", that returns this variable. The test just used this function.
+
+
+What is the expected retrun value? It's 350. We assigned to it 350 in the *constructor*, the function that runs once,
```
Comment (pipermerriam) [inline docs/dev_cycle.part-03.rst]: s/retrun/return/

File: docs/dev_cycle.part-04.rst [hunk 30]
```diff
@@ -0,0 +1,315 @@
...
+    def test_default_usd_rate(chain):
+        donator, deploy_tx_hash = chain.provider.get_or_deploy_contract('Donator')
+        default_usd_rate = donator.call().default_usd_rate()
+        assert default_usd_rate == 350
+
+
+    def test_donate(chain):
+        donator, deploy_tx_hash = chain.provider.get_or_deploy_contract('Donator')
+        donator.transact({'value':500}).donate(370)
+        donator.transact({'value':650}).donate(380)
```
Comment (pipermerriam) [inline docs/dev_cycle.part-04.rst]: This should be modified to include saving the transaction hash and waiting for the transaction to be mined since this will only pass on a `tester` chain as it is.

File: docs/dev_cycle.part-04.rst [hunk 31]
```diff
@@ -0,0 +1,315 @@
...
+
+**[6] Asserts**: We expect the ``donations_total`` to be ``500 + 650 = 1150``, the ``donations_count`` is 2,
+and the ``default_usd_rate`` to match the last update, 380. The test gets the varaibles with ``call``, but in the case
+of a "getter" it doesn't matter, since a "getter" function will not change the contract's state wether it's called with ``transact`` or ``call``.
+
+Run the test:
+
+.. code-block:: bash
+
+    $ py.test --disable-pytest-warnings
```
Comment (pipermerriam) [inline docs/dev_cycle.part-04.rst]: Why the `--disable-pytest-warnings`?

File: docs/dev_cycle.part-04.rst [hunk 32]
```diff
@@ -0,0 +1,315 @@
...
+
+    $ nano tests/test_donator.py
+
+Add the following test to the bottom of the file:
+
+.. code-block:: python
+
+    def test_usd_calculation(chain):
+
+        ONE_ETH_IN_WEI = 10**18  # 1 ETH == 1,000,000,000,000,000,000 Wei
```
Comment (pipermerriam) [inline docs/dev_cycle.part-04.rst]: Use web3 for this.  `web3.toWei(1, 'ether')`

File: docs/dev_cycle.part-04.rst [hunk 33]
```diff
@@ -0,0 +1,315 @@
...
+The test runs two transactions: note the ``transact`` function, which will change the contract state and balance
+on the blockchain. We use the ``tester`` chain, so the state is reset on each test run.
+
+**First transaction**
+
+.. code-block:: python
+
+    donator.transact({'value':ONE_ETH_IN_WEI}).donate(400)
+
+Donate Wei worth of 1 Ether, where the effective ETH/USD rate is $400. That is, $400 per Ether,
```
Comment (pipermerriam) [inline docs/dev_cycle.part-04.rst]: Rather than using `$400` which is both above current market value, and could be interpreted as a price projection, can we use a less real number like $5 which is way below market value and more of a *made-up* number?

File: docs/dev_cycle.part-04.rst [hunk 34]
```diff
@@ -0,0 +1,315 @@
...
+.. code-block:: shell
+
+    $ nano contracts/Donator.sol
+
+
+And fix the line to:
+
+.. code-block:: solidity
+
+    uint in_usd = msg.value * usd_rate / 10**18;
```
Comment (pipermerriam) [inline docs/dev_cycle.part-04.rst]: instead of `10**18` you can use a literal of `1 ether` and solidity will do the math for you.

File: docs/dev_cycle.part-04.rst [hunk 35]
```diff
@@ -0,0 +1,315 @@
...
+
+The contract seems Ok, but to be on the safe side, we will run next a few tests for the edge cases.
+
+
+
+
+
+
+
+
```
Comment (pipermerriam) [inline docs/dev_cycle.part-04.rst]: Bonus whitespace!

File: docs/quickstart.rst [hunk 36]
```diff
@@ -144,6 +144,7 @@ Alternatively, you can init a new project by a directory:
 
     $ populus -p /path/to/my/project/ init
 
+.. _greete_quickstart:
```
Comment (pipermerriam) [inline docs/quickstart.rst]: s/greete/greeter

File: docs/dev_cycle.intro.rst (hunk 3 above)
Comment (Aviah) [reply docs/dev_cycle.intro.rst]: ok

File: docs/dev_cycle.intro.rst (hunk 4 above)
Comment (Aviah) [reply docs/dev_cycle.intro.rst]: ok

File: docs/dev_cycle.intro.rst (hunk 5 above)
Comment (Aviah) [reply docs/dev_cycle.intro.rst]: ok

File: docs/dev_cycle.part-01.rst (hunk 8 above)
Comment (Aviah) [reply docs/dev_cycle.part-01.rst]: ok 0.4.11

File: docs/dev_cycle.part-01.rst (hunk 10 above)
Comment (Aviah) [reply docs/dev_cycle.part-01.rst]: ok

File: docs/dev_cycle.part-01.rst (hunk 16 above)
Comment (Aviah) [reply docs/dev_cycle.part-01.rst]: ok

File: docs/dev_cycle.part-03.rst (hunk 25 above)
Comment (Aviah) [reply docs/dev_cycle.part-03.rst]: ok, summery in one sent

File: docs/dev_cycle.part-03.rst (hunk 29 above)
Comment (Aviah) [reply docs/dev_cycle.part-03.rst]: ok

File: docs/dev_cycle.part-04.rst (hunk 30 above)
Comment (Aviah) [reply docs/dev_cycle.part-04.rst]: will add in another place later, to keep things simple at the stage

File: docs/dev_cycle.part-04.rst (hunk 31 above)
Comment (Aviah) [reply docs/dev_cycle.part-04.rst]: unrelated warnings which may be confusing. will add a note

File: docs/dev_cycle.part-04.rst (hunk 32 above)
Comment (Aviah) [reply docs/dev_cycle.part-04.rst]: explained later (part 8), here it's more intuitive

File: docs/dev_cycle.part-04.rst (hunk 33 above)
Comment (Aviah) [reply docs/dev_cycle.part-04.rst]: ok

File: docs/dev_cycle.part-04.rst (hunk 34 above)
Comment (Aviah) [reply docs/dev_cycle.part-04.rst]: ok

File: docs/quickstart.rst (hunk 36 above)
Comment (Aviah) [reply docs/quickstart.rst]: ok
